### PR TITLE
8.0 sale option

### PR DESCRIPTION
**Pull request by Viggor**
[WIP] Add optionnal product in sale order line.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you explain a bit more the purpose of the module? There's no description of it.

File: sale_option/README.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+  :alt: License: AGPL-3
+
+Sale Option
+==============
+
+This module provide an options system for sale order line.
```
Comment (sebastienbeau) [inline sale_option/README.rst:7]: @Viggor  Can you update the description in order to explain to @pedrobaeza  and everyone the purpose of the module

File: sale_option/sale.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    @api.onchange('bom_line_id', 'qty')
+    @api.depends('bom_line_id', 'qty')
+    def _compute_price(self):
+        option_price = 0
+        if self.bom_line_id and self.sale_line_id.pricelist_id:
+            option_price = self.sale_line_id.pricelist_id.with_context(
+                {
+                    'uom': self.bom_line_id.product_uom.id,
+                    'date': self.sale_line_id.order_id.date_order,
+                    'with_rm': self.sale_line_id.price_with_material,
```
Comment (sebastienbeau) [inline sale_option/sale.py:118]: This code should be not here, can you fix it?
You can add a method that return a custom context

File: sale_option/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
```
Comment (Cedric-Pigeon) [inline sale_option/__openerp__.py:21]: Please use simple headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: sale_option/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+{'name': 'Sale option',
+ 'version': '0.0.1',
+ 'author': 'Akretion',
```
Comment (Cedric-Pigeon) [inline sale_option/__openerp__.py:25]: Please add Odoo Community Association (OCA) as author

File: sale_option/mrp.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
```
Comment (Cedric-Pigeon) [inline sale_option/mrp.py:21]: Same remark for header

File: sale_option/mrp.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+class BomOption(models.Model):
+    _name = "mrp.bom.line.option"
+
+    @api.model
+    def _get_type(self):
+        selection = (
+            ('select', 'selection'),
+            ('multiselect', 'multi-selection'),
+            ('required', 'Required'),
+        )
```
Comment (Cedric-Pigeon) [inline sale_option/mrp.py:35]: Please capitalize first letter for option description

File: sale_option/mrp_view.xml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        <record id="view_bom_option_form" model="ir.ui.view">
+            <field name="model">mrp.bom.line.option</field>
+            <field name="arch" type="xml">
+                <form string="BoM Option">
+                    <sheet>
+                        <group col="6">
+                        <field name="name" select="1"/>
+                        <field name="sequence"/>
+                        <field name="type"/>
+                </group>
```
Comment (Cedric-Pigeon) [inline sale_option/mrp_view.xml:25]: Please fix indent

File: sale_option/sale.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
```
Comment (Cedric-Pigeon) [inline sale_option/sale.py:21]: Same remark for header

**Comment by Cedric-Pigeon** [comment]
@Viggor Thanks for the contrib but could you better organize files in your module. All xml view files in a view directory, python files in a model directory, ...

**Comment by pedrobaeza** [comment]
No answer, so closing. Feel free to reopen if still needed.

**Pull request closed without merging**